### PR TITLE
  Update spotless to meet JDK-21 baseline

### PR DESCRIPTION
**Pull request by reta**
**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Add JDK-21 to Github actions

**Describe the solution you are proposing**

**Describe alternatives you've considered**

**Additional context**

### Check List
- [x] Backport Labels added.   
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@ansjcy purely formatting updates (spotless update is needed to work with JDK-21)

**Review by ansjcy — APPROVED** [review]
LGTM.

File: src/test/java/org/opensearch/performanceanalyzer/net/NetClientTest.java [hunk 1]
```diff
@@ -34,6 +35,7 @@
 @PowerMockIgnore({"org.apache.logging.log4j.*", "com.sun.org.apache.xerces.*"})
 @RunWith(PowerMockRunner.class)
 @PrepareForTest({NetClient.class, InterNodeRpcServiceGrpc.InterNodeRpcServiceStub.class})
+@Ignore("See please https://github.com/opensearch-project/performance-analyzer-rca/issues/529")
```
Comment (reta) [inline src/test/java/org/opensearch/performanceanalyzer/net/NetClientTest.java:38]: @ansjcy we need to work on PowerMock removal, https://github.com/opensearch-project/performance-analyzer-rca/issues/529, it is not playing well with recent JDKs (and is dead essentially)

**Review by getsaurabh02 — DISMISSED** [review]

**Comment by prudhvigodithi** [comment]
Looks like this is the change required for 2.12 builds to pass, https://github.com/opensearch-project/performance-analyzer/issues/611, can we please get this in soon and backport to 2.x?

**Review by bbarani — APPROVED** [review]

**Review by prudhvigodithi — APPROVED** [review]

**Pull request merged**